### PR TITLE
DX-2851 Major Cypress Refactoring

### PR DESCRIPTION
**Pull request by brianluisgomez**
**⚠️Please do not make any changes to spec files in this repository!⚠️**

**All spec changes should be done in the [api-specs](https://github.com/Bandwidth/api-specs) repository.**

**Please follow the [Guide](https://bandwidth-jira.atlassian.net/wiki/spaces/DX/pages/4098359409/How+to+Update+API+Specifications+and+Contribute+to+Developer+Documentation) for contributing to our developer documentation.**

### CODE REVIEW

**Review by ajrice6713 — APPROVED** [review]

**Review by ckoegel — APPROVED** [review]

File: site/cypress/e2e/tests/search.cy.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    it('search by operation', () => {
+        cy.get('.menu-content [role="search"]').type('List');
+    })
+
+    it('search by tag', () => {
+        cy.get('.menu-content [role="search"]').type('Media');
+    })
+
+    it('search by endpoint', () => {
+        cy.get('.menu-content [role="search"]').type('/messages');
```
Comment (bpateldx) [inline site/cypress/e2e/tests/search.cy.js:52]: Validate what we expect after typing text in the search box! Hint: you can review ReDoc E2E Tests

**Review by bpateldx — CHANGES_REQUESTED** [review]
The Tests suite looks way better after refactor - the ease of maintenance and support has increased to great extent. Left one comment for review otherwise looks really good.

File: site/cypress/e2e/tests/search.cy.js (hunk 1 above)
Comment (brianluisgomez) [reply site/cypress/e2e/tests/search.cy.js]: I refactored the search test a bit in utils. So now it'll also quickly validate that what we typed in is in fact in the search bar. Let me know if this is what you meant.

File: site/cypress/utils/utils.js [hunk 2]
```diff
@@ -117,3 +117,60 @@ export const navTester = (path) => {
...
+})
+
+it('type in searchbar', () => {
+    cy.get('#docsearch-input').type(`${searchText}`);
+    cy.get('.DocSearch-Dropdown-Container').children().should('have.length.greaterThan', 1);
+    cy.get('.DocSearch-Hit-source').first().should('contain', `${searchText}`);
+    cy.get('.DocSearch-HitsFooter > a').invoke('text').should('match', /See all \d+ results/);
+    cy.get('.DocSearch-Reset').click();
+    cy.get('#docsearch-input').should('have.text', '');
+    cy.get('#docsearch-input').should('have.attr', 'placeholder', 'Search docs');
```
Comment (bpateldx) [inline site/cypress/utils/utils.js:158]: 👍

**Review by bpateldx — APPROVED** [review]
Covering few Test Cases for Searches would save us from finding out breaking changes on future upgrades. Thank you for adding it.

File: site/cypress/e2e/tests/search.cy.js (hunk 1 above)
Comment (bpateldx) [reply site/cypress/e2e/tests/search.cy.js]: Looks good, great work on quick updates.

**Pull request merged**